lib/ui/text/paragraph_impl_txt.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/text/paragraph_impl_txt.h"
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/text/paragraph.h"
+#include "flutter/lib/ui/text/paragraph_impl.h"
+#include "lib/ftl/tasks/task_runner.h"
+
+#include "lib/txt/src/paragraph_constraints.h"
+
+namespace blink {
+
+ParagraphImplTxt::ParagraphImplTxt(std::unique_ptr<txt::Paragraph>& paragraph) {
+  m_paragraph = std::move(paragraph);
+}
+
+ParagraphImplTxt::~ParagraphImplTxt() {}
+
+double ParagraphImplTxt::width() {
+  return m_width;
+}
+
+double ParagraphImplTxt::height() {  // TODO.
+  return m_paragraph->GetHeight();
+}
+
+double ParagraphImplTxt::minIntrinsicWidth() {  // TODO.
+  return FLT_MAX;
+}
+
+double ParagraphImplTxt::maxIntrinsicWidth() {  // TODO.
+  return FLT_MAX;
+}
+
+double ParagraphImplTxt::alphabeticBaseline() {  // TODO.
+  return m_paragraph->GetAlphabeticBaseline();
+}
+
+double ParagraphImplTxt::ideographicBaseline() {  // TODO.
+  return m_paragraph->GetIdeographicBaseline();
+}
+
+bool ParagraphImplTxt::didExceedMaxLines() {
+  return m_paragraph->DidExceedMaxLines();
+}
+
+void ParagraphImplTxt::layout(double width) {
+  m_width = width;
+  m_paragraph->Layout(txt::ParagraphConstraints{width});
```

Comment:
I'd remove ParagraphConstraints from libtxt.  It doesn't really serve a purpose.  I commented on that in a previous CL, but it looks like it wasn't ever removed.